### PR TITLE
Project finished!

### PR DESCRIPTION
**Pull request by joaquinauc**
It certainly was somewhat of a challenge, I'm really having fun doing these!

Please be brutal in your code review, and thank you for taking the time to review it!

### CODE REVIEW

File: Calculator.JoaquinUrrea/Calculator/CalculatorFunctionality.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
+﻿using CalculatorLibrary;
+namespace CalculatorProgram;
+
+internal static class CalculatorFunctionality
+{
```
Comment (nwdorian) [inline Calculator.JoaquinUrrea/Calculator/CalculatorFunctionality.cs:5]: 🟠 **Static classes**

❓If we make everything `static`, we are throwing away all the OOP concepts, as we no longer work with class instances (objects).

➡️ As a rule of thumb, avoid the `static` keyword completely and only use it if you really need it. You can do just about anything and everything without using `static` at all, so it shouldn't be needed.

File: Calculator.JoaquinUrrea/Calculator/CalculatorInterface.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            else if (calculatorOption == CalculatorOption.LatestCalculations)
+            {
+                CalculatorFunctionality.ShowLatestHistory();
+            }
+            else
+            {
+                CalculatorFunctionality.CalculatorLogic();
+            }
+
+        } while (exitProgram == false);
```
Comment (nwdorian) [inline Calculator.JoaquinUrrea/Calculator/CalculatorInterface.cs:36]: 🟡 **Code style**

❓Avoid using `if (isValid == false)` since the variable is already a boolean.

➡️ Since `isValid` is a boolean, explicitly comparing it to `false` is redundant and against C# conventions. The correct and required form is to use the negation operator:
```suggestion

        } while (!exitProgram);
```

File: Calculator.JoaquinUrrea/Calculator/CalculatorInterface.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+        do
+        {
+            Console.Clear();
+
+            calculatorOption = AnsiConsole.Prompt(
+                new SelectionPrompt<CalculatorOption>()
+                .Title("Select which action you wish to do: ")
+                .AddChoices(Enum.GetValues<CalculatorOption>())
+            );
```
Comment (nwdorian) [inline Calculator.JoaquinUrrea/Calculator/CalculatorInterface.cs:21]: 🟡 **Human readable enums**

Using enums makes the magic strings go away and simplifies the code base, but it is not the best user experience. Look into the `[DisplayName]` attribute and use a [converter](https://spectreconsole.net/api/spectre.console/selectionprompt_1/c3f97978) with `Spectre.Console` to show a human friendly version. e.g. `Add Session`.

**Review by nwdorian — APPROVED** [review]
Hey @joaquinauc 👋,

Excellent work on your **Calculator** project submission! 🎉

I have finished the code review and left some feedback below. ☺️

Feel free to reach out if you have questions or if you'd like to collaborate on any of these points.✌️

---

🟢 **Requirements**

⭐ You have completed all of the project requirements!

---

🏅 **Extra challenges**

✅ Count the amount of times the calculator was used
✅ List of latest calculations
✅ Use previous results for new calculations
✅ Add extra operations
💪 Very nice to see you implemented extra challenges as it speeds up the learning process! 

---

🟠 **Argument exception**

❓Select `LastestCalculations` > continue > `UseResult` and `Delete` both throw an argument exception if the collection is empty. 

image

➡️ Make sure to thoroughly test the application before submitting for review. Cover happy and failure paths, invalid inputs and edge cases. Code reviews get stricter in future projects and you might be asked to fix issues like this.



---

🟡 **UX improvements**

➡️ It would be nice to have an **Exit** option inside the `Calculator` menu instead of the "press 'n' to return to menu" only after calculation was done.

---

💭 You can take care of issues in your own repo if you want, but since the project will be approved you don't have to push the changes or create a new PR.

✅ I will go ahead and approve the project submission. Keep up the good work on the next ones! 😄

Best regards,  
@nwdorian
